### ISSUE TITLE
When opening a shared mailbox in OWA by "Open Another Mailbox" item.getsharedPropertiesAsync is undefined

### ISSUE THREAD

**Issue by GrumpyAL** [OfficeDev/office-js#1295]
> Message from office-js bot:
We’re closing this issue because it has been inactive for a long time. We’re doing this to keep the issues list manageable and useful for everyone. If this issue is still relevant for you, please create a new issue.
> Thank you for your understanding and continued feedback.

When using OWA to open a shared mailbox by selecting "Open Another Mailbox" we are unable to determine that we are in a delegate scenario / shared folder

## Expected Behavior
item.getsharedPropertiesAsync returns the same was when accessing a shared item through a different method (e.g. having the shared mailbox or folder open on the left-hand side within your own mailbox.

## Current Behavior

- item.getsharedPropertiesAsync is undefined
- REST token returned is for logged in account, not mailbox _(would be expected in a delegate scenario normally)_
- userprofile.displayName and userprofile.emailaddress is to the shared box _(makes it seem like we are acting as/on-behalf of this user/mailbox rather than accessing it as a delegate or shared folder)_

This causes problems when we try to construct the REST url on the backend using /me because we were unable to detect that we were in a shared mailbox.  /me returns a 404 as expected when the token used is for a different mailbox (As described in  #1146)

## Steps to Reproduce, or Live Example

1. Load a manifest with the supportssharedfolders element set to true.
2. Deploy the manifest via centralized deployment or manually to a shared mailbox.
3. Open the shared mailbox in OWA by going to your user icon and selecting "Open Another Mailbox."
4. Invoke the add-in
5. In the console, attempt to call the item.getsharedPropertiesAsync function

e.g. 
```
Office.context.mailbox.item.getSharedPropertiesAsync(function(result) {
  var sharedproperties = result.value;
console.log(sharedproperties);
console.log(sharedproperties.targetRestUrl);
});
```

## Context
Is this behavior intended? I appear to be unable to check that I'm in a delegate scenario here.  Is it expected to always construct the REST URL using the details found here? https://docs.microsoft.com/en-us/javascript/api/outlook/office.sharedproperties?view=outlook-js-1.8

The documentation for office.context.mailbox.restUrl shows using the /v2.0/me endpoint, but that appears to not work when accessing a shared mailbox as described above given that the token is issued to the logged in user and not the mailbox that is open.  I understand when we are in a delegate scenario we need to use /users instead of /me but I'm not sure if we should be using /users all the time to account for the above when we can't be sure if we're in a delegate scenario or not (or is there another way to detect?).

So my main question is, should we construct our REST URL as below to account for all possible situations?
1. Shared folder opend from within the user's mailbox
2. Shared mailbox opened via OWA "Open Another Mailbox"
3. User in their own mailbox.

When getsharedPropertiesAsync is undefined _(Situations 2 and 3)_:
Office.context.mailbox.restUrl + "/{api_version}/users/" + Office.context.mailbox.userprofile.emailAddress + "/{REST_operation}"

when we have return from item.getsharedPropertiesAsync _(Situation 1)_
targetRestUrl + "/{api_version}/users/" + targetMailbox + "/{REST_operation}"

Would we expect constructing the REST URL in this way to work for all possible shared scenarios?  This appears to work in my testing.  I just want to make sure this isn't a bug or related to #1140 and that we are intended to construct all REST URLs using /user (even if we think we're not in a delegate scenario) when delegate scenarios are possible for our add-in.

To summarize:
1. Should this scenario be returning values for item.getsharedPropertiesAsync? Or...
2. Should the token being issued in this scenario be for that mailbox and not the logged in user? Or...
3. Should we construct REST URLs in the way described above?

Thank you for your time, as always.

## Your Environment

* Platform: Outlook on the Web
* Host: Outlook
* Operating System: Windows 10
* Browser (if using Office on the web): Chrome

**Comment by exextoc** [comment]
Delegate add-ins scenario are currently undefined for shared mailboxes that are accessed through 'Open another mailbox'. This is in our product backlog and we will update our documentation when its available.

**Comment by GrumpyAL** [comment]
> Delegate add-ins scenario are currently undefined for shared mailboxes that are accessed through 'Open another mailbox'. This is in our product backlog and we will update our documentation when its available.

Hello, @exextoc!  Just wondering if there is any rough release estimate on this enhancement.  Also, which specific documentation page should I watch for an update? Would it be here (https://docs.microsoft.com/en-us/office/dev/add-ins/outlook/delegate-access) or elsewhere? Thanks so much!
